Bio/SeqIO/Interfaces.py

```diff
@@ -58,21 +58,17 @@ def __init__(self, source, alphabet=None, mode="t", fmt=None):
             else:
                 raise ValueError(f"Unknown mode '{mode}'") from None
             self.stream = source
-            self.should_close_stream = False
         try:
             self.records = self.parse(self.stream)
         except Exception:
-            if self.should_close_stream:
-                self.stream.close()
+            self.stream.close()
             raise
 
     def __next__(self):
         """Return the next entry."""
         try:
             return next(self.records)
         except Exception:
-            if self.should_close_stream:
-                self.stream.close()
             raise
```

Comment:
What is the purpose of the `try:` `except:` block?

Reply:
I think if the record is empty then, it might generate error  and that's why I add ``try:`` ``except:`` block

Reply:
Yes, end of file and no more records does seem likely to trigger this try/except.